### DISCUSSION TITLE
Support for in place encryption / decryption

### DISCUSSION THREAD

**Discussion by ghost** [FiloSottile/age#508] [General]
Hi together,

currently it is not directly possible to encrypt / decrypt a file in place, i.e. replacing the file with it's encrypted / unencrypted representation.
If someone is defining the input file also as an output file, it leads to a corrupt file and it looks like all data are unrecoverable lost. Please see issue #491 for this.

Therefore and to simplify the ux I recommend to define a flag that allows in place encryption / decryption.
Sure, this means that both representations have to be kept in memory which could be a problem if someone encrypts / decrypts a large file. But i think in the most cases this would be a useful addition.

Regards

**Comment by Turmaxx** [comment]
this is something I would also like to see added to age as well.

**Reply by ghost** [reply]
I implemented it and pushed a pull request but unfortunately no reaction...
https://github.com/FiloSottile/age/pull/523

**Reply by Turmaxx** [reply]
@iFrozenPhoenix  thanks for implementing this feature
hopefully, he sees it since it would be amazing instead of having to use output files.
